src/listener.js

```diff
@@ -0,0 +1,129 @@
+'use strict'
+
+const multiaddr = require('multiaddr')
+const debug = require('debug')
+const log = debug('libp2p:tcp')
+const Connection = require('interface-connection').Connection
+const os = require('os')
+const contains = require('lodash.contains')
+const net = require('net')
+const toPull = require('stream-to-pull-stream')
+
+const getMultiaddr = require('./get-multiaddr')
+
+const IPFS_CODE = 421
+const CLOSE_TIMEOUT = 2000
+
+module.exports = (handler) => {
+  const listener = net.createServer((socket) => {
```

Comment:
It is missing to bubble up the events from the listener:
- [ ] event: 'listening'
- [ ] event: 'close'
- [ ] event: 'connection'
- [ ] event: 'error'

See interface-transport.

- [ ] Add tests for the above